epoxy-processor/src/main/java/com/airbnb/epoxy/ControllerModelField.kt

```diff
@@ -0,0 +1,36 @@
+package com.airbnb.epoxy
+
+import com.squareup.javapoet.TypeName
+
+import javax.lang.model.element.Element
+
+import com.airbnb.epoxy.Utils.isFieldPackagePrivate
+
+class ControllerModelField(element: Element) {
+
+    var fieldName: String = element.simpleName.toString()
+    var typeName: TypeName = TypeName.get(element.asType())
+    var packagePrivate: Boolean = isFieldPackagePrivate(element)
+
+    override fun equals(other: Any?): Boolean {
```

Comment:
@geralt-encore Ah right. What if it is a data class with a private constructor and a companion with a factory method to build the class based on an Element?

I'm not sure if that is a common pattern, but it seems a little more elegant than manually implementing everything.

I'll go ahead and merge now just to get it in, but I'm curious what you think about that pattern ^